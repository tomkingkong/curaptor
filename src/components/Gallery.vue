<template>
  <div class="backdrop">
    <h1>{{ msg }}</h1>
    <div class="canvas">
      <ul class="cards">
        <li v-for="(card, index) in cards" :key="index">
          <img :src="card.primaryimageurl" width="400" class="painting"/>
        </li>
      </ul>
    </div>
    <!-- <input type="text" v-model="input_val">
    <div>
      Input Value: <span v-text="input_val"></span>
    </div>
    <div>
      <button class="btn btn-primary" v-on:click="counter++">
        You've clicked this button {{counter}} times!
      </button>
    </div> -->
  </div>
</template>

<script>
export default {
  name: 'Gallery',
  data () {
    return {
      msg: 'CurAptor',
      input_val: '',
      counter: 0
    }
  },
  props: {
    cards: Array
  }
}
</script>

<style scoped>
  .canvas {
    background: rgb(250, 250, 250);
    box-shadow: 0rem 0rem 0.5rem rgb(160, 160, 160);
    min-height: 100vh;
    height: 100%;
    width: 96%;
    padding: 0.5rem;
    margin: 0 auto;
  }
  .cards {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    list-style-type: none;
    list-style-position: outside;
    padding: 0;
  }
  .painting {
    background: lightgrey;
    border-radius: 0.2rem;
    max-width: 13rem;
    transition: all 0.15s linear;
  }
  li {
    min-width: 10rem;
    width: fit-content;
    height: fit-content;
    margin: 1rem 0.5rem 0 0.5rem;
    -webkit-column-break-inside: avoid;
    page-break-inside: avoid;
    break-inside: avoid;
    transition: all 0.15s linear;
  }
  li:hover {
    box-shadow: 0.1rem 0.1rem 0.5rem 0.01rem rgb(194, 194, 194);
  }
  li:hover > .painting{
    max-width: 20rem;
    width: 20rem;
  }
</style>
