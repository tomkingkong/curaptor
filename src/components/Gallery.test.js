import {shallow} from 'vue-test-utils'
import Gallery from './Gallery.vue'

describe('Gallery', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(Gallery)
    expect(wrapper).toMatchSnapshot()
  })
})