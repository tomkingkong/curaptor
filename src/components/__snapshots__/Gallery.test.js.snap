// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery matches snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "element": <div
    class="backdrop"
  >
    <div
      class="canvas"
    >
      <ul
        class="cards"
      />
    </div>
  </div>,
  "isVueComponent": true,
  "options": Object {
    "attachedToDocument": false,
  },
  "update": [Function],
  "version": 2.5,
  "vm": VueComponent {
    "$_mountingOptionsSlots": undefined,
    "$_originalSlots": Object {},
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$el": <div
      class="backdrop"
    >
      <div
        class="canvas"
      >
        <ul
          class="cards"
        />
      </div>
    </div>,
    "$emit": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_Ctor": Object {
        "1": [Function],
      },
      "_base": [Function],
      "_parentElm": null,
      "_propKeys": Array [
        "cards",
      ],
      "_refElm": null,
      "beforeCreate": Array [
        [Function],
      ],
      "components": Object {
        "transition": Object {
          "render": [Function],
        },
        "transition-group": Object {
          "render": [Function],
        },
      },
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "name": "Gallery",
      "props": Object {
        "cards": Object {
          "type": [Function],
        },
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "__emitted": Object {},
    "__emittedByOrder": Array [],
    "_c": [Function],
    "_data": Object {
      "seen": false,
    },
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": true,
    "_isVue": true,
    "_props": Object {
      "cards": undefined,
    },
    "_renderProxy": VueComponent {
      "$_mountingOptionsSlots": undefined,
      "$_originalSlots": Object {},
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        class="backdrop"
      >
        <div
          class="canvas"
        >
          <ul
            class="cards"
          />
        </div>
      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "1": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_propKeys": Array [
          "cards",
        ],
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "render": [Function],
          },
          "transition-group": Object {
            "render": [Function],
          },
        },
        "data": [Function],
        "directives": Object {},
        "filters": Object {},
        "name": "Gallery",
        "props": Object {
          "cards": Object {
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_data": Object {
        "seen": false,
      },
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_props": Object {
        "cards": undefined,
      },
      "_renderProxy": [Circular],
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": Array [
          VNode {
            "asyncFactory": undefined,
            "asyncMeta": undefined,
            "children": Array [
              VNode {
                "asyncFactory": undefined,
                "asyncMeta": undefined,
                "children": undefined,
                "componentInstance": undefined,
                "componentOptions": undefined,
                "context": [Circular],
                "data": Object {
                  "staticClass": "cards",
                },
                "elm": <ul
                  class="cards"
                />,
                "fnContext": undefined,
                "fnOptions": undefined,
                "fnScopeId": undefined,
                "isAsyncPlaceholder": false,
                "isCloned": false,
                "isComment": false,
                "isOnce": false,
                "isRootInsert": false,
                "isStatic": false,
                "key": undefined,
                "ns": undefined,
                "parent": undefined,
                "raw": false,
                "tag": "ul",
                "text": undefined,
              },
            ],
            "componentInstance": undefined,
            "componentOptions": undefined,
            "context": [Circular],
            "data": Object {
              "staticClass": "canvas",
            },
            "elm": <div
              class="canvas"
            >
              <ul
                class="cards"
              />
            </div>,
            "fnContext": undefined,
            "fnOptions": undefined,
            "fnScopeId": undefined,
            "isAsyncPlaceholder": false,
            "isCloned": false,
            "isComment": false,
            "isOnce": false,
            "isRootInsert": false,
            "isStatic": false,
            "key": undefined,
            "ns": undefined,
            "parent": undefined,
            "raw": false,
            "tag": "div",
            "text": undefined,
          },
        ],
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": [Circular],
        "data": Object {
          "staticClass": "backdrop",
        },
        "elm": <div
          class="backdrop"
        >
          <div
            class="canvas"
          >
            <ul
              class="cards"
            />
          </div>
        </div>,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": true,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "div",
        "text": undefined,
      },
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {
          2,
        },
        "deps": Array [
          Dep {
            "id": 2,
            "subs": Array [
              [Circular],
            ],
          },
        ],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {
            2,
          },
          "deps": Array [
            Dep {
              "id": 2,
              "subs": Array [
                [Circular],
              ],
            },
          ],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 1,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
      "seen": false,
    },
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": VNode {
      "asyncFactory": undefined,
      "asyncMeta": undefined,
      "children": Array [
        VNode {
          "asyncFactory": undefined,
          "asyncMeta": undefined,
          "children": Array [
            VNode {
              "asyncFactory": undefined,
              "asyncMeta": undefined,
              "children": undefined,
              "componentInstance": undefined,
              "componentOptions": undefined,
              "context": [Circular],
              "data": Object {
                "staticClass": "cards",
              },
              "elm": <ul
                class="cards"
              />,
              "fnContext": undefined,
              "fnOptions": undefined,
              "fnScopeId": undefined,
              "isAsyncPlaceholder": false,
              "isCloned": false,
              "isComment": false,
              "isOnce": false,
              "isRootInsert": false,
              "isStatic": false,
              "key": undefined,
              "ns": undefined,
              "parent": undefined,
              "raw": false,
              "tag": "ul",
              "text": undefined,
            },
          ],
          "componentInstance": undefined,
          "componentOptions": undefined,
          "context": [Circular],
          "data": Object {
            "staticClass": "canvas",
          },
          "elm": <div
            class="canvas"
          >
            <ul
              class="cards"
            />
          </div>,
          "fnContext": undefined,
          "fnOptions": undefined,
          "fnScopeId": undefined,
          "isAsyncPlaceholder": false,
          "isCloned": false,
          "isComment": false,
          "isOnce": false,
          "isRootInsert": false,
          "isStatic": false,
          "key": undefined,
          "ns": undefined,
          "parent": undefined,
          "raw": false,
          "tag": "div",
          "text": undefined,
        },
      ],
      "componentInstance": undefined,
      "componentOptions": undefined,
      "context": [Circular],
      "data": Object {
        "staticClass": "backdrop",
      },
      "elm": <div
        class="backdrop"
      >
        <div
          class="canvas"
        >
          <ul
            class="cards"
          />
        </div>
      </div>,
      "fnContext": undefined,
      "fnOptions": undefined,
      "fnScopeId": undefined,
      "isAsyncPlaceholder": false,
      "isCloned": false,
      "isComment": false,
      "isOnce": false,
      "isRootInsert": true,
      "isStatic": false,
      "key": undefined,
      "ns": undefined,
      "parent": undefined,
      "raw": false,
      "tag": "div",
      "text": undefined,
    },
    "_watcher": Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {
        2,
      },
      "deps": Array [
        Dep {
          "id": 2,
          "subs": Array [
            [Circular],
          ],
        },
      ],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 1,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": false,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
    "_watchers": Array [
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {
          2,
        },
        "deps": Array [
          Dep {
            "id": 2,
            "subs": Array [
              [Circular],
            ],
          },
        ],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    ],
    "seen": false,
  },
  "vnode": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": Array [
      VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": Array [
          VNode {
            "asyncFactory": undefined,
            "asyncMeta": undefined,
            "children": undefined,
            "componentInstance": undefined,
            "componentOptions": undefined,
            "context": VueComponent {
              "$_mountingOptionsSlots": undefined,
              "$_originalSlots": Object {},
              "$attrs": Object {},
              "$children": Array [],
              "$createElement": [Function],
              "$el": <div
                class="backdrop"
              >
                <div
                  class="canvas"
                >
                  <ul
                    class="cards"
                  />
                </div>
              </div>,
              "$emit": [Function],
              "$listeners": Object {},
              "$options": Object {
                "_Ctor": Object {
                  "1": [Function],
                },
                "_base": [Function],
                "_parentElm": null,
                "_propKeys": Array [
                  "cards",
                ],
                "_refElm": null,
                "beforeCreate": Array [
                  [Function],
                ],
                "components": Object {
                  "transition": Object {
                    "render": [Function],
                  },
                  "transition-group": Object {
                    "render": [Function],
                  },
                },
                "data": [Function],
                "directives": Object {},
                "filters": Object {},
                "name": "Gallery",
                "props": Object {
                  "cards": Object {
                    "type": [Function],
                  },
                },
                "render": [Function],
                "staticRenderFns": Array [],
              },
              "$parent": undefined,
              "$refs": Object {},
              "$root": [Circular],
              "$scopedSlots": Object {},
              "$slots": Object {},
              "$vnode": undefined,
              "__emitted": Object {},
              "__emittedByOrder": Array [],
              "_c": [Function],
              "_data": Object {
                "seen": false,
              },
              "_directInactive": false,
              "_events": Object {},
              "_hasHookEvent": false,
              "_inactive": null,
              "_isBeingDestroyed": false,
              "_isDestroyed": false,
              "_isMounted": true,
              "_isVue": true,
              "_props": Object {
                "cards": undefined,
              },
              "_renderProxy": VueComponent {
                "$_mountingOptionsSlots": undefined,
                "$_originalSlots": Object {},
                "$attrs": Object {},
                "$children": Array [],
                "$createElement": [Function],
                "$el": <div
                  class="backdrop"
                >
                  <div
                    class="canvas"
                  >
                    <ul
                      class="cards"
                    />
                  </div>
                </div>,
                "$emit": [Function],
                "$listeners": Object {},
                "$options": Object {
                  "_Ctor": Object {
                    "1": [Function],
                  },
                  "_base": [Function],
                  "_parentElm": null,
                  "_propKeys": Array [
                    "cards",
                  ],
                  "_refElm": null,
                  "beforeCreate": Array [
                    [Function],
                  ],
                  "components": Object {
                    "transition": Object {
                      "render": [Function],
                    },
                    "transition-group": Object {
                      "render": [Function],
                    },
                  },
                  "data": [Function],
                  "directives": Object {},
                  "filters": Object {},
                  "name": "Gallery",
                  "props": Object {
                    "cards": Object {
                      "type": [Function],
                    },
                  },
                  "render": [Function],
                  "staticRenderFns": Array [],
                },
                "$parent": undefined,
                "$refs": Object {},
                "$root": [Circular],
                "$scopedSlots": Object {},
                "$slots": Object {},
                "$vnode": undefined,
                "__emitted": Object {},
                "__emittedByOrder": Array [],
                "_c": [Function],
                "_data": Object {
                  "seen": false,
                },
                "_directInactive": false,
                "_events": Object {},
                "_hasHookEvent": false,
                "_inactive": null,
                "_isBeingDestroyed": false,
                "_isDestroyed": false,
                "_isMounted": true,
                "_isVue": true,
                "_props": Object {
                  "cards": undefined,
                },
                "_renderProxy": [Circular],
                "_self": [Circular],
                "_staticTrees": null,
                "_uid": 0,
                "_vnode": [Circular],
                "_watcher": Watcher {
                  "active": true,
                  "cb": [Function],
                  "deep": false,
                  "depIds": Set {
                    2,
                  },
                  "deps": Array [
                    Dep {
                      "id": 2,
                      "subs": Array [
                        [Circular],
                      ],
                    },
                  ],
                  "dirty": false,
                  "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                  "getter": [Function],
                  "id": 1,
                  "lazy": false,
                  "newDepIds": Set {},
                  "newDeps": Array [],
                  "sync": false,
                  "user": false,
                  "value": undefined,
                  "vm": [Circular],
                },
                "_watchers": Array [
                  Watcher {
                    "active": true,
                    "cb": [Function],
                    "deep": false,
                    "depIds": Set {
                      2,
                    },
                    "deps": Array [
                      Dep {
                        "id": 2,
                        "subs": Array [
                          [Circular],
                        ],
                      },
                    ],
                    "dirty": false,
                    "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                    "getter": [Function],
                    "id": 1,
                    "lazy": false,
                    "newDepIds": Set {},
                    "newDeps": Array [],
                    "sync": false,
                    "user": false,
                    "value": undefined,
                    "vm": [Circular],
                  },
                ],
                "seen": false,
              },
              "_self": [Circular],
              "_staticTrees": null,
              "_uid": 0,
              "_vnode": [Circular],
              "_watcher": Watcher {
                "active": true,
                "cb": [Function],
                "deep": false,
                "depIds": Set {
                  2,
                },
                "deps": Array [
                  Dep {
                    "id": 2,
                    "subs": Array [
                      [Circular],
                    ],
                  },
                ],
                "dirty": false,
                "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                "getter": [Function],
                "id": 1,
                "lazy": false,
                "newDepIds": Set {},
                "newDeps": Array [],
                "sync": false,
                "user": false,
                "value": undefined,
                "vm": [Circular],
              },
              "_watchers": Array [
                Watcher {
                  "active": true,
                  "cb": [Function],
                  "deep": false,
                  "depIds": Set {
                    2,
                  },
                  "deps": Array [
                    Dep {
                      "id": 2,
                      "subs": Array [
                        [Circular],
                      ],
                    },
                  ],
                  "dirty": false,
                  "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                  "getter": [Function],
                  "id": 1,
                  "lazy": false,
                  "newDepIds": Set {},
                  "newDeps": Array [],
                  "sync": false,
                  "user": false,
                  "value": undefined,
                  "vm": [Circular],
                },
              ],
              "seen": false,
            },
            "data": Object {
              "staticClass": "cards",
            },
            "elm": <ul
              class="cards"
            />,
            "fnContext": undefined,
            "fnOptions": undefined,
            "fnScopeId": undefined,
            "isAsyncPlaceholder": false,
            "isCloned": false,
            "isComment": false,
            "isOnce": false,
            "isRootInsert": false,
            "isStatic": false,
            "key": undefined,
            "ns": undefined,
            "parent": undefined,
            "raw": false,
            "tag": "ul",
            "text": undefined,
          },
        ],
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": VueComponent {
          "$_mountingOptionsSlots": undefined,
          "$_originalSlots": Object {},
          "$attrs": Object {},
          "$children": Array [],
          "$createElement": [Function],
          "$el": <div
            class="backdrop"
          >
            <div
              class="canvas"
            >
              <ul
                class="cards"
              />
            </div>
          </div>,
          "$emit": [Function],
          "$listeners": Object {},
          "$options": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "_base": [Function],
            "_parentElm": null,
            "_propKeys": Array [
              "cards",
            ],
            "_refElm": null,
            "beforeCreate": Array [
              [Function],
            ],
            "components": Object {
              "transition": Object {
                "render": [Function],
              },
              "transition-group": Object {
                "render": [Function],
              },
            },
            "data": [Function],
            "directives": Object {},
            "filters": Object {},
            "name": "Gallery",
            "props": Object {
              "cards": Object {
                "type": [Function],
              },
            },
            "render": [Function],
            "staticRenderFns": Array [],
          },
          "$parent": undefined,
          "$refs": Object {},
          "$root": [Circular],
          "$scopedSlots": Object {},
          "$slots": Object {},
          "$vnode": undefined,
          "__emitted": Object {},
          "__emittedByOrder": Array [],
          "_c": [Function],
          "_data": Object {
            "seen": false,
          },
          "_directInactive": false,
          "_events": Object {},
          "_hasHookEvent": false,
          "_inactive": null,
          "_isBeingDestroyed": false,
          "_isDestroyed": false,
          "_isMounted": true,
          "_isVue": true,
          "_props": Object {
            "cards": undefined,
          },
          "_renderProxy": VueComponent {
            "$_mountingOptionsSlots": undefined,
            "$_originalSlots": Object {},
            "$attrs": Object {},
            "$children": Array [],
            "$createElement": [Function],
            "$el": <div
              class="backdrop"
            >
              <div
                class="canvas"
              >
                <ul
                  class="cards"
                />
              </div>
            </div>,
            "$emit": [Function],
            "$listeners": Object {},
            "$options": Object {
              "_Ctor": Object {
                "1": [Function],
              },
              "_base": [Function],
              "_parentElm": null,
              "_propKeys": Array [
                "cards",
              ],
              "_refElm": null,
              "beforeCreate": Array [
                [Function],
              ],
              "components": Object {
                "transition": Object {
                  "render": [Function],
                },
                "transition-group": Object {
                  "render": [Function],
                },
              },
              "data": [Function],
              "directives": Object {},
              "filters": Object {},
              "name": "Gallery",
              "props": Object {
                "cards": Object {
                  "type": [Function],
                },
              },
              "render": [Function],
              "staticRenderFns": Array [],
            },
            "$parent": undefined,
            "$refs": Object {},
            "$root": [Circular],
            "$scopedSlots": Object {},
            "$slots": Object {},
            "$vnode": undefined,
            "__emitted": Object {},
            "__emittedByOrder": Array [],
            "_c": [Function],
            "_data": Object {
              "seen": false,
            },
            "_directInactive": false,
            "_events": Object {},
            "_hasHookEvent": false,
            "_inactive": null,
            "_isBeingDestroyed": false,
            "_isDestroyed": false,
            "_isMounted": true,
            "_isVue": true,
            "_props": Object {
              "cards": undefined,
            },
            "_renderProxy": [Circular],
            "_self": [Circular],
            "_staticTrees": null,
            "_uid": 0,
            "_vnode": [Circular],
            "_watcher": Watcher {
              "active": true,
              "cb": [Function],
              "deep": false,
              "depIds": Set {
                2,
              },
              "deps": Array [
                Dep {
                  "id": 2,
                  "subs": Array [
                    [Circular],
                  ],
                },
              ],
              "dirty": false,
              "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
              "getter": [Function],
              "id": 1,
              "lazy": false,
              "newDepIds": Set {},
              "newDeps": Array [],
              "sync": false,
              "user": false,
              "value": undefined,
              "vm": [Circular],
            },
            "_watchers": Array [
              Watcher {
                "active": true,
                "cb": [Function],
                "deep": false,
                "depIds": Set {
                  2,
                },
                "deps": Array [
                  Dep {
                    "id": 2,
                    "subs": Array [
                      [Circular],
                    ],
                  },
                ],
                "dirty": false,
                "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
                "getter": [Function],
                "id": 1,
                "lazy": false,
                "newDepIds": Set {},
                "newDeps": Array [],
                "sync": false,
                "user": false,
                "value": undefined,
                "vm": [Circular],
              },
            ],
            "seen": false,
          },
          "_self": [Circular],
          "_staticTrees": null,
          "_uid": 0,
          "_vnode": [Circular],
          "_watcher": Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {
              2,
            },
            "deps": Array [
              Dep {
                "id": 2,
                "subs": Array [
                  [Circular],
                ],
              },
            ],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 1,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": false,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
          "_watchers": Array [
            Watcher {
              "active": true,
              "cb": [Function],
              "deep": false,
              "depIds": Set {
                2,
              },
              "deps": Array [
                Dep {
                  "id": 2,
                  "subs": Array [
                    [Circular],
                  ],
                },
              ],
              "dirty": false,
              "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
              "getter": [Function],
              "id": 1,
              "lazy": false,
              "newDepIds": Set {},
              "newDeps": Array [],
              "sync": false,
              "user": false,
              "value": undefined,
              "vm": [Circular],
            },
          ],
          "seen": false,
        },
        "data": Object {
          "staticClass": "canvas",
        },
        "elm": <div
          class="canvas"
        >
          <ul
            class="cards"
          />
        </div>,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": false,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "div",
        "text": undefined,
      },
    ],
    "componentInstance": undefined,
    "componentOptions": undefined,
    "context": VueComponent {
      "$_mountingOptionsSlots": undefined,
      "$_originalSlots": Object {},
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        class="backdrop"
      >
        <div
          class="canvas"
        >
          <ul
            class="cards"
          />
        </div>
      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "1": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_propKeys": Array [
          "cards",
        ],
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "render": [Function],
          },
          "transition-group": Object {
            "render": [Function],
          },
        },
        "data": [Function],
        "directives": Object {},
        "filters": Object {},
        "name": "Gallery",
        "props": Object {
          "cards": Object {
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_data": Object {
        "seen": false,
      },
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_props": Object {
        "cards": undefined,
      },
      "_renderProxy": VueComponent {
        "$_mountingOptionsSlots": undefined,
        "$_originalSlots": Object {},
        "$attrs": Object {},
        "$children": Array [],
        "$createElement": [Function],
        "$el": <div
          class="backdrop"
        >
          <div
            class="canvas"
          >
            <ul
              class="cards"
            />
          </div>
        </div>,
        "$emit": [Function],
        "$listeners": Object {},
        "$options": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "_base": [Function],
          "_parentElm": null,
          "_propKeys": Array [
            "cards",
          ],
          "_refElm": null,
          "beforeCreate": Array [
            [Function],
          ],
          "components": Object {
            "transition": Object {
              "render": [Function],
            },
            "transition-group": Object {
              "render": [Function],
            },
          },
          "data": [Function],
          "directives": Object {},
          "filters": Object {},
          "name": "Gallery",
          "props": Object {
            "cards": Object {
              "type": [Function],
            },
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "$parent": undefined,
        "$refs": Object {},
        "$root": [Circular],
        "$scopedSlots": Object {},
        "$slots": Object {},
        "$vnode": undefined,
        "__emitted": Object {},
        "__emittedByOrder": Array [],
        "_c": [Function],
        "_data": Object {
          "seen": false,
        },
        "_directInactive": false,
        "_events": Object {},
        "_hasHookEvent": false,
        "_inactive": null,
        "_isBeingDestroyed": false,
        "_isDestroyed": false,
        "_isMounted": true,
        "_isVue": true,
        "_props": Object {
          "cards": undefined,
        },
        "_renderProxy": [Circular],
        "_self": [Circular],
        "_staticTrees": null,
        "_uid": 0,
        "_vnode": [Circular],
        "_watcher": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {
            2,
          },
          "deps": Array [
            Dep {
              "id": 2,
              "subs": Array [
                [Circular],
              ],
            },
          ],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 1,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        "_watchers": Array [
          Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {
              2,
            },
            "deps": Array [
              Dep {
                "id": 2,
                "subs": Array [
                  [Circular],
                ],
              },
            ],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 1,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": false,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
        ],
        "seen": false,
      },
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": [Circular],
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {
          2,
        },
        "deps": Array [
          Dep {
            "id": 2,
            "subs": Array [
              [Circular],
            ],
          },
        ],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": false,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {
            2,
          },
          "deps": Array [
            Dep {
              "id": 2,
              "subs": Array [
                [Circular],
              ],
            },
          ],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 1,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": false,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
      "seen": false,
    },
    "data": Object {
      "staticClass": "backdrop",
    },
    "elm": <div
      class="backdrop"
    >
      <div
        class="canvas"
      >
        <ul
          class="cards"
        />
      </div>
    </div>,
    "fnContext": undefined,
    "fnOptions": undefined,
    "fnScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": false,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": "div",
    "text": undefined,
  },
}
`;
